src/ServiceInsight/MessageList/MessageListViewModel.cs

```diff
@@ -286,5 +293,15 @@ public void BringIntoView(StoredMessage msg)
         {
             eventAggregator.PublishOnUIThread(new ScrollDiagramItemIntoView(msg));
         }
+
+        public void OnSavePartLayout()
+        {
+            view.OnSaveLayout(settingsProvider);
+        }
+
+        void RestoreLayout()
+        {
+            view.OnRestoreLayout(settingsProvider);
+        }
```

Comment:
I see that `RestoreLayout` is called from `OnViewAttached`. Is there an equivalent method that could call `OnSavePartLayout`, to remove the need for the `IPersistPartLayout` interface, and its explicit call from `ShellViewModel`?

Reply:
The ViewAttach stuff comes from `IViewAware` interface of caliburn. It doesn't have anything for the view is detached.

The interface `IPersistPartLayout` still would be required though, as it is an opt-in interface.